src/client/src/ui/spotTile/SpotTile.tsx

```diff
@@ -82,72 +54,59 @@ export default class SpotTile extends React.Component<SpotTileProps, {}> {
       'spot-tile--error': hasNotification && notification.hasError,
     })
 
-    const newWindowClassName = classnames('popout__controls  glyphicon glyphicon-new-window', {
-      'spot-tile__icon--tearoff': !canPopout,
-      'spot-tile__icon--hidden': canPopout,
-    })
-    const spotTileContent = (
-      <div>
-        <span className="spot-tile__execution-label">Executing</span>
-        {priceComponents}
-        <NotionalContainer
-          className={notionalInputClass}
-          currencyPair={currencyPair}
-        />
-        <div className={spotDateClass}>
-          <span className="spot-tile__tenor">SP</span>
-          <span className="spot-tile__delivery-date">. {formattedDate}</span>
-        </div>
-      </div>
-    )
+    return className
+  }
+
+  createSpotTileControls() {
+    const { onPopoutClick, undockTile, displayCurrencyChart, isRunningInOpenFin, spotTileData } = this.props
+
+    const spotTileControlsProps = {
+      onPopoutClick,
+      undockTile,
+      displayCurrencyChart,
+      isRunningInOpenFin,
+      currencyChartIsOpening: spotTileData.currencyChartIsOpening
+    }
     return (
-      <div className={className}>
-        <div className="spot-tile__container">
-          <div className="spot-tile__controls">
-            <i className={chartIQIconClassName}
-               onClick={() => displayCurrencyChart()}/>
-            <i className={newWindowClassName}
-               onClick={() => onPopoutClick()}/>
-            <i className="popout__undock spot-tile__icon--undock glyphicon glyphicon-log-out"
-               onClick={() => undockTile()}/>
-          </div>
-          {!hasNotification ? spotTileContent : generatedNotification}
-        </div>
-      </div>
+      <SpotTileControls { ...spotTileControlsProps }/>
```

Comment:
sometimes spreading props can be a bit of a code smell. Makes it hard to understand what the components contract is.